app/code/Magento/MediaGalleryCatalogUi/Ui/Component/Listing/Columns/Thumbnail.php

```diff
@@ -64,7 +72,7 @@ public function prepareDataSource(array $dataSource)
                 if (isset($item[$fieldName])) {
                     $item[$fieldName . '_src'] = $this->getUrl($item[$fieldName]);
                 } else {
-                    $category = new DataObject($item);
```

Comment:
Hey, @jmonteros422 , after a closer look the approach appears to be completely invalid: this class is responsible to prepare the image for category, not for the product, so `CategoryFactory` and `Category\ImageFactory` should be used. Also we have to find out how to get a placeholder image in case category does not have one assigned.

Here is how the method should look like based on my understanding:

```
    public function prepareDataSource(array $dataSource)
    {
        if (!isset($dataSource['data']['items'])) {
            return $dataSource;
        }

        $fieldName = $this->getData('name');
        foreach ($dataSource['data']['items'] as & $item) {
            if (isset($item[$fieldName])) {
                $item[$fieldName . '_src'] = $this->getUrl($item[$fieldName]);
                continue;
            }

            if (isset($item['entity_id'])) {
                $src = $this->imageFactory->create()->getUrl(
                    $this->categoryRepository->get($item['entity_id'])
                );
                $item[$fieldName . '_src'] = $src;
                continue;
            }

            $item[$fieldName . '_src'] = $defaultImage;
        }
    }
```

Reply:
Ok Sergii, will check that one out. Thank you!